Sources/NIOFileSystem/FileHandleProtocol.swift

```diff
@@ -376,33 +376,31 @@ extension ReadableFileHandleProtocol {
                 forceChunkedRead = true
             }
 
-            if !forceChunkedRead, readSize <= singleShotReadLimit {
-                return try await self.readChunk(
-                    fromAbsoluteOffset: offset,
-                    length: .bytes(Int64(readSize))
-                )
+            let chunkLength: ByteCount = if !forceChunkedRead, readSize <= singleShotReadLimit {
+                .bytes(Int64(readSize))
```

Comment:
We still support Swift 5.8 so you can't use this syntax, you'll need to declare `let chunkLength: ByteCount` and then assign to it.